ethers-solc/src/artifacts.rs

```diff
@@ -537,12 +537,17 @@ impl CompilerOutput {
     }
 
     /// Whether the output contains a compiler warning
-    pub fn has_warning(&self) -> bool {
-        self.errors.iter().any(|err| err.severity.is_warning())
+    pub fn has_warning(&self, ignored_error_codes: &[u64]) -> bool {
+        self.errors.iter().any(|err| {
+            let is_ignored =
+                err.error_code.as_ref().map_or(false, |code| !ignored_error_codes.contains(code));
```

Comment:
I'd be more efficient to compute this only if `err.severity.is_warning() == true`